main-app/app/scripts/providers/game-model.js

```diff
@@ -0,0 +1,126 @@
+(function () {
+    'use strict';
+
+    angular.module('Tombola.NoughtsAndCrosses.Providers')
+        .factory('GameModel',['$http', '$timeout', '$location', 'Proxy', 'Audio', function($http, $timeout, $location, proxy, audio) {
+            var Model = function() {
+                var me = this;
+
+                me.currentGrid = [[{key: "0", value: "0"}, {key: "1", value: "0"}, {key: "2", value: "0"}],
+                    [{key: "3", value: "0"}, {key: "4", value: "0"}, {key: "5", value: "0"}],
+                    [{key: "6", value: "0"}, {key: "7", value: "0"}, {key: "8", value: "0"}]];
+
+                me.playerTypes = [{name: "HUMAN", value: "human"}, {name: "EASY", value: "random"}, {
+                    name: "HARD",
+                    value: "pre-trained"
+                }];
+                me.player1Type = "";
+                me.player2Type = "";
+
+                me.playerCross = 1;
+                me.playerNought = 2;
+                me.player1Icon = "Crosses";
+
+                me.gameOver = true; //possibly not actually used anywhere, but its set at the right times so might be useful
+                me.showWinner = false;
+                me.winnerText = "";
+
+                me.isPlayerTypeValid = function(playerNum){
+                    if(playerNum===1){
+                        for (var i = 0; i < me.playerTypes.length; i++) {
```

Comment:
Really finnicky - declare i (and j below) at the top of this function - this is its actual scope and doing this makes it clear. Then initialise it to 0 in the loop.
Also declare me.playerTypes.length as a variable and use that - some JS implementations will keep accessing the object rather evaluating once (as in c#) - this is perf hit This variable needs to be declared at the top of the function also.
I've had both of these points picked up in code I've written - it's more of a style thing (if you ignore the performance) and this is the style the team has adopted.